SickChill 20240301 test

BKSteve opened the pull request:
## Update to 2024.3.1

<!--Major update to SickChill for 2024 -->

Fixes many things. 
Updates packages and removes unused.
Aligns requirements-crossenv with latest Python 3.11.5-8

## Checklist

- [x] Build rule `all-supported` completed successfully
- [ ] New installation of package completed successfully
- [ ] Package upgrade completed successfully (Manually install the package again)
- [ ] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Package-Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created


### Type of change

<!--Please use any relavent tags.-->
- [x] Bug fix
- [ ] New Package
- [x] Package update
- [ ] Includes small framework changes
- [ ] This change requires a documentation update (e.g. Wiki)

---
**Do not merge**

CorentinWicht commented:
@BKSteve is it possible to already test this 20240301 test package? 

Best,

C.

BKSteve commented:
Yes you can. 

https://discord.com/channels/502612977271439372/1208779268570615828

BKSteve commented:
@th0ma7  @hgy59 I notice the build system wasted it's time making an `88f6281-6.2.4` with Python inside.
Also all others have Python too, this seems like a significant overhead, any way to stop this from my side?

hgy59 commented:
> Also all others have Python in addition to SickChill too, this seems like a significant overhead, any way to stop this from my side?

This is not a significat overhead. Python needs to be built anyway to build packages that depend on python. Since we reuse prebuilt python for such packages, the github build action creates the python package first.

If you build sickchill locally, you can benefit from prebuilt python, if you build python spk before sickchill. While development you can rebuild sickchill (make clean and make arch-*) without recreating python. This saves a lot of time when building multiple packages that depend on python, but even for a single package you save time if you have to build a package more than once.

We added the reuse of prebuilt dependent packages for packages depending on ffmpeg and packages depending on python.

BKSteve commented:
Any advice on fixing qoriq build failure? How to disable/avoid qoriq?
I'll link to synocli-file https://github.com/SynoCommunity/spksrc/pull/6288 so I know when there's a solution.

mreid-tt commented:
> Any advice on fixing qoriq build failure? How to disable/avoid qoriq? I'll link to synocli-file #6288 so I know when there's a solution.

There was a recent fix with #6307. Can you try re-basing this PR to see if it addresses the qoriq build?

mreid-tt commented:
@BKSteve, I've completed the rebase for you to resolve the qoriq build issue. Please complete the testing checklist to verify that the PR is ready for merging.

@hgy59 any other fixes required?

mreid-tt commented:
@BKSteve, I tried to run a clean install but got the following error:

```
2024/11/10 09:12:43	install sickchill 20240301-11 Begin preinst
2024/11/10 09:12:43	Begin initialize_variables
2024/11/10 09:12:43	End initialize_variables
2024/11/10 09:12:43	===> Step preinst. STATUS=INSTALL USER=sickchill GROUP=sc-download SHARE_PATH=
2024/11/10 09:12:43	install sickchill 20240301-11 End preinst ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mkdir -p /volume1/@appstore/sickchill -m 755
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mkdir -p /volume1/@appstore/sickchill -m 755 ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/rm -rf /volume1/@appstore/sickchill
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/rm -rf /volume1/@appstore/sickchill ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/package /volume1/@appstore/sickchill
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/package /volume1/@appstore/sickchill ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/rm -rf /var/packages/sickchill
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/rm -rf /var/packages/sickchill ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mkdir -p /var/packages/sickchill -m 755
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mkdir -p /var/packages/sickchill -m 755 ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/touch /var/packages/sickchill/installing
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/touch /var/packages/sickchill/installing ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/INFO /var/packages/sickchill/INFO
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/INFO /var/packages/sickchill/INFO ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/rm -rf /var/packages/sickchill/scripts
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/rm -rf /var/packages/sickchill/scripts ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/scripts /var/packages/sickchill/scripts
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/scripts /var/packages/sickchill/scripts ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/rm -rf /var/packages/sickchill/WIZARD_UIFILES
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/rm -rf /var/packages/sickchill/WIZARD_UIFILES ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/WIZARD_UIFILES /var/packages/sickchill/WIZARD_UIFILES
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/WIZARD_UIFILES /var/packages/sickchill/WIZARD_UIFILES ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/rm -rf /var/packages/sickchill/conf
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/rm -rf /var/packages/sickchill/conf ret=[0]
2024/11/10 09:12:46	install sickchill 20240301-11 Begin /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/conf /var/packages/sickchill/conf
2024/11/10 09:12:46	install sickchill 20240301-11 End /bin/mv -f /volume1/@tmp/synopkg/install.0nYYeI/conf /var/packages/sickchill/conf ret=[0]
2024/11/10 09:13:07	install sickchill 20240301-11 Begin postinst
2024/11/10 09:13:07	Begin initialize_variables
2024/11/10 09:13:07	End initialize_variables
2024/11/10 09:13:07	===> Step postinst. STATUS=INSTALL USER=sickchill GROUP=sc-download SHARE_PATH=
2024/11/10 09:13:07	Begin save_wizard_variables
2024/11/10 09:13:07	End save_wizard_variables
2024/11/10 09:13:07	Begin syno_sync_var_folder
2024/11/10 09:13:07	End syno_sync_var_folder
2024/11/10 09:13:07	Begin service_postinst
2024/11/10 09:13:07	-----------------------------------------------
2024/11/10 09:13:07	Python 3.11.10
2024/11/10 09:13:16	Default pip version 24.2
2024/11/10 09:13:37	-----------------------------------------------
2024/11/10 09:13:37	Install packages from wheels
2024/11/10 09:13:37	Install packages from wheels [/volume1/@appstore/sickchill/share/wheelhouse/requirements.txt]
2024/11/10 09:13:39	Looking in links: /volume1/@appstore/sickchill/share/wheelhouse
2024/11/10 09:13:43	Collecting CacheControl==0.14.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 1))
2024/11/10 09:13:43	  Downloading cachecontrol-0.14.0-py3-none-any.whl.metadata (3.1 kB)
2024/11/10 09:13:50	Collecting Deprecated==1.2.14 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 2))
2024/11/10 09:13:50	  Downloading Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)
2024/11/10 09:13:56	Collecting Flask==3.0.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 3))
2024/11/10 09:13:56	  Downloading flask-3.0.3-py3-none-any.whl.metadata (3.2 kB)
2024/11/10 09:13:59	Collecting Mako==1.3.5 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 4))
2024/11/10 09:14:00	  Downloading Mako-1.3.5-py3-none-any.whl.metadata (2.9 kB)
2024/11/10 09:14:04	Collecting MarkupSafe==2.1.5 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 5))
2024/11/10 09:14:04	  Downloading MarkupSafe-2.1.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)
2024/11/10 09:14:09	Collecting PyGithub==2.4.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 6))
2024/11/10 09:14:09	  Downloading PyGithub-2.4.0-py3-none-any.whl.metadata (3.9 kB)
2024/11/10 09:14:15	Collecting PyJWT==2.9.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 7))
2024/11/10 09:14:15	  Downloading PyJWT-2.9.0-py3-none-any.whl.metadata (3.0 kB)
2024/11/10 09:14:25	Collecting PyNMA==1.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 8))
2024/11/10 09:14:25	  Downloading PyNMA-1.0.tar.gz (2.4 kB)
2024/11/10 09:14:31	  Preparing metadata (setup.py): started
2024/11/10 09:14:32	  Preparing metadata (setup.py): finished with status 'done'
2024/11/10 09:14:39	Processing /volume1/@appstore/sickchill/share/wheelhouse/PyNaCl-1.5.0-cp311-cp311-linux_x86_64.whl (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 9))
2024/11/10 09:14:44	Collecting SQLAlchemy==2.0.32 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 10))
2024/11/10 09:14:44	  Downloading SQLAlchemy-2.0.32-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.6 kB)
2024/11/10 09:14:53	Collecting Send2Trash==1.8.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 11))
2024/11/10 09:14:53	  Downloading Send2Trash-1.8.3-py3-none-any.whl.metadata (4.0 kB)
2024/11/10 09:15:01	Collecting Unidecode==1.3.8 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 12))
2024/11/10 09:15:01	  Downloading Unidecode-1.3.8-py3-none-any.whl.metadata (13 kB)
2024/11/10 09:15:04	Collecting appdirs==1.4.4 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 13))
2024/11/10 09:15:04	  Downloading appdirs-1.4.4-py2.py3-none-any.whl.metadata (9.0 kB)
2024/11/10 09:15:09	Collecting attrs==24.2.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 14))
2024/11/10 09:15:09	  Downloading attrs-24.2.0-py3-none-any.whl.metadata (11 kB)
2024/11/10 09:15:13	Collecting babelfish==0.6.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 15))
2024/11/10 09:15:13	  Downloading babelfish-0.6.1-py3-none-any.whl.metadata (3.1 kB)
2024/11/10 09:15:18	Collecting beautifulsoup4==4.12.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 16))
2024/11/10 09:15:18	  Downloading beautifulsoup4-4.12.3-py3-none-any.whl.metadata (3.8 kB)
2024/11/10 09:15:29	Collecting beekeeper-alt==2022.9.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 17))
2024/11/10 09:15:29	  Downloading beekeeper_alt-2022.9.3-py3-none-any.whl.metadata (5.3 kB)
2024/11/10 09:15:33	Collecting bencode.py==4.0.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 18))
2024/11/10 09:15:33	  Downloading bencode.py-4.0.0-py2.py3-none-any.whl.metadata (5.8 kB)
2024/11/10 09:15:37	Collecting cffi==1.17.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 19))
2024/11/10 09:15:37	  Downloading cffi-1.17.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.5 kB)
2024/11/10 09:15:41	Collecting chardet==5.2.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 20))
2024/11/10 09:15:41	  Downloading chardet-5.2.0-py3-none-any.whl.metadata (3.4 kB)
2024/11/10 09:15:46	Collecting charset-normalizer==3.4.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 21))
2024/11/10 09:15:46	  Downloading charset_normalizer-3.4.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (34 kB)
2024/11/10 09:15:49	Collecting cinemagoer==2023.5.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 22))
2024/11/10 09:15:49	  Downloading cinemagoer-2023.5.1-py3-none-any.whl.metadata (2.9 kB)
2024/11/10 09:15:53	Collecting click==8.1.7 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 23))
2024/11/10 09:15:53	  Downloading click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
2024/11/10 09:15:59	Collecting configobj==5.0.9 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 24))
2024/11/10 09:15:59	  Downloading configobj-5.0.9.tar.gz (101 kB)
2024/11/10 09:16:03	     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 101.5/101.5 kB 26.7 kB/s eta 0:00:00
2024/11/10 09:16:03	  Installing build dependencies: started
2024/11/10 09:16:31	  Installing build dependencies: finished with status 'done'
2024/11/10 09:16:31	  Getting requirements to build wheel: started
2024/11/10 09:16:32	  Getting requirements to build wheel: finished with status 'done'
2024/11/10 09:16:32	  Preparing metadata (pyproject.toml): started
2024/11/10 09:16:33	  Preparing metadata (pyproject.toml): finished with status 'done'
2024/11/10 09:16:35	Collecting cryptography==41.0.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 25))
2024/11/10 09:16:35	  Downloading cryptography-41.0.3-cp37-abi3-manylinux_2_28_x86_64.whl.metadata (5.2 kB)
2024/11/10 09:16:42	Collecting decorator==5.1.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 26))
2024/11/10 09:16:42	  Downloading decorator-5.1.1-py3-none-any.whl.metadata (4.0 kB)
2024/11/10 09:16:48	Collecting deluge-client==1.10.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 27))
2024/11/10 09:16:48	  Downloading deluge_client-1.10.2-py3-none-any.whl.metadata (4.3 kB)
2024/11/10 09:16:56	Collecting dogpile.cache==1.3.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 28))
2024/11/10 09:16:56	  Downloading dogpile.cache-1.3.3-py3-none-any.whl.metadata (5.2 kB)
2024/11/10 09:17:03	Collecting enzyme==0.5.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 29))
2024/11/10 09:17:03	  Downloading enzyme-0.5.2-py3-none-any.whl.metadata (2.5 kB)
2024/11/10 09:17:09	Collecting future==1.0.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 30))
2024/11/10 09:17:09	  Downloading future-1.0.0-py3-none-any.whl.metadata (4.0 kB)
2024/11/10 09:17:14	Collecting gntp==1.0.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 31))
2024/11/10 09:17:14	  Downloading gntp-1.0.3-py3-none-any.whl.metadata (4.5 kB)
2024/11/10 09:17:19	Collecting greenlet==3.0.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 32))
2024/11/10 09:17:19	  Downloading greenlet-3.0.3-cp311-cp311-manylinux_2_24_x86_64.manylinux_2_28_x86_64.whl.metadata (3.8 kB)
2024/11/10 09:17:22	Collecting guessit==3.8.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 33))
2024/11/10 09:17:22	  Downloading guessit-3.8.0-py3-none-any.whl.metadata (140 kB)
2024/11/10 09:17:24	     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.7/140.7 kB 79.1 kB/s eta 0:00:00
2024/11/10 09:17:25	Collecting idna==3.10 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 34))
2024/11/10 09:17:25	  Downloading idna-3.10-py3-none-any.whl.metadata (10 kB)
2024/11/10 09:17:31	Collecting ifaddr==0.2.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 35))
2024/11/10 09:17:31	  Downloading ifaddr-0.2.0-py3-none-any.whl.metadata (4.9 kB)
2024/11/10 09:17:35	Collecting imagesize==1.4.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 36))
2024/11/10 09:17:35	  Downloading imagesize-1.4.1-py2.py3-none-any.whl.metadata (1.5 kB)
2024/11/10 09:17:42	Collecting ipaddress==1.0.23 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 37))
2024/11/10 09:17:42	  Downloading ipaddress-1.0.23-py2.py3-none-any.whl.metadata (923 bytes)
2024/11/10 09:17:48	Collecting jsonrpclib-pelix==0.4.3.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 38))
2024/11/10 09:17:48	  Downloading jsonrpclib_pelix-0.4.3.3-py2.py3-none-any.whl.metadata (18 kB)
2024/11/10 09:17:55	Collecting kodipydent-alt==2022.9.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 39))
2024/11/10 09:17:56	  Downloading kodipydent_alt-2022.9.3-py3-none-any.whl.metadata (3.7 kB)
2024/11/10 09:18:07	Collecting lxml==5.3.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 40))
2024/11/10 09:18:07	  Downloading lxml-5.3.0-cp311-cp311-manylinux_2_28_x86_64.whl.metadata (3.8 kB)
2024/11/10 09:18:11	Collecting markdown2==2.5.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 41))
2024/11/10 09:18:11	  Downloading markdown2-2.5.1-py2.py3-none-any.whl.metadata (2.2 kB)
2024/11/10 09:18:14	Collecting mock==5.1.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 42))
2024/11/10 09:18:14	  Downloading mock-5.1.0-py3-none-any.whl.metadata (3.0 kB)
2024/11/10 09:18:19	Collecting msgpack==1.0.7 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 43))
2024/11/10 09:18:19	  Downloading msgpack-1.0.7-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.1 kB)
2024/11/10 09:18:29	Collecting new-rtorrent-python==1.0.1a0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 44))
2024/11/10 09:18:29	  Downloading new_rtorrent_python-1.0.1a0-py3-none-any.whl.metadata (798 bytes)
2024/11/10 09:18:38	Collecting oauthlib==3.2.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 45))
2024/11/10 09:18:38	  Downloading oauthlib-3.2.2-py3-none-any.whl.metadata (7.5 kB)
2024/11/10 09:18:43	Collecting packaging==24.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 46))
2024/11/10 09:18:43	  Downloading packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
2024/11/10 09:18:47	Collecting pbr==6.1.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 47))
2024/11/10 09:18:47	  Downloading pbr-6.1.0-py2.py3-none-any.whl.metadata (3.4 kB)
2024/11/10 09:18:51	Collecting pluggy==1.5.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 48))
2024/11/10 09:18:51	  Downloading pluggy-1.5.0-py3-none-any.whl.metadata (4.8 kB)
2024/11/10 09:18:55	Collecting profilehooks==1.13.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 49))
2024/11/10 09:18:55	  Downloading profilehooks-1.13.0-py3-none-any.whl.metadata (7.0 kB)
2024/11/10 09:18:59	Collecting putio.py==8.7.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 50))
2024/11/10 09:18:59	  Downloading putio.py-8.7.0.tar.gz (7.3 kB)
2024/11/10 09:19:01	  Preparing metadata (setup.py): started
2024/11/10 09:19:02	  Preparing metadata (setup.py): finished with status 'done'
2024/11/10 09:19:04	Collecting pyOpenSSL==23.2.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 51))
2024/11/10 09:19:04	  Downloading pyOpenSSL-23.2.0-py3-none-any.whl.metadata (10 kB)
2024/11/10 09:19:10	Collecting pycparser==2.22 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 52))
2024/11/10 09:19:10	  Downloading pycparser-2.22-py3-none-any.whl.metadata (943 bytes)
2024/11/10 09:19:18	Collecting pymediainfo==6.1.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 53))
2024/11/10 09:19:18	  Downloading pymediainfo-6.1.0.tar.gz (446 kB)
2024/11/10 09:19:20	     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 446.5/446.5 kB 178.5 kB/s eta 0:00:00
2024/11/10 09:19:20	  Preparing metadata (setup.py): started
2024/11/10 09:19:55	  Preparing metadata (setup.py): finished with status 'done'
2024/11/10 09:19:59	Collecting pyparsing==3.2.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 54))
2024/11/10 09:19:59	  Downloading pyparsing-3.2.0-py3-none-any.whl.metadata (5.0 kB)
2024/11/10 09:20:08	Collecting pysrt==1.1.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 55))
2024/11/10 09:20:08	  Downloading pysrt-1.1.2.tar.gz (104 kB)
2024/11/10 09:20:14	     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 104.4/104.4 kB 17.8 kB/s eta 0:00:00
2024/11/10 09:20:14	  Preparing metadata (setup.py): started
2024/11/10 09:20:15	  Preparing metadata (setup.py): finished with status 'done'
2024/11/10 09:20:20	Collecting pytest==8.3.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 56))
2024/11/10 09:20:20	  Downloading pytest-8.3.3-py3-none-any.whl.metadata (7.5 kB)
2024/11/10 09:20:27	Collecting python-dateutil==2.9.0.post0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 57))
2024/11/10 09:20:27	  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
2024/11/10 09:20:30	Collecting python-slugify==8.0.4 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 58))
2024/11/10 09:20:30	  Downloading python_slugify-8.0.4-py2.py3-none-any.whl.metadata (8.5 kB)
2024/11/10 09:20:39	Collecting python-twitter==3.5 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 59))
2024/11/10 09:20:39	  Downloading python_twitter-3.5-py2.py3-none-any.whl.metadata (32 kB)
2024/11/10 09:20:42	Collecting python3-fanart==2.0.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 60))
2024/11/10 09:20:42	  Downloading python3_fanart-2.0.0-py3-none-any.whl.metadata (4.8 kB)
2024/11/10 09:20:51	Collecting pytz==2024.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 61))
2024/11/10 09:20:51	  Downloading pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
2024/11/10 09:20:59	Collecting qbittorrent-api==2024.9.67 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 62))
2024/11/10 09:20:59	  Downloading qbittorrent_api-2024.9.67-py3-none-any.whl.metadata (20 kB)
2024/11/10 09:21:13	Collecting rarfile==4.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 63))
2024/11/10 09:21:13	  Downloading rarfile-4.2-py3-none-any.whl.metadata (4.4 kB)
2024/11/10 09:21:23	Collecting rebulk==3.2.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 64))
2024/11/10 09:21:23	  Downloading rebulk-3.2.0-py3-none-any.whl.metadata (22 kB)
2024/11/10 09:21:28	Collecting requests-oauthlib==2.0.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 65))
2024/11/10 09:21:29	  Downloading requests_oauthlib-2.0.0-py2.py3-none-any.whl.metadata (11 kB)
2024/11/10 09:21:34	Collecting requests==2.32.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 66))
2024/11/10 09:21:34	  Downloading requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
2024/11/10 09:21:39	Collecting sickchill==2024.3.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 67))
2024/11/10 09:21:39	  Downloading sickchill-2024.3.1-py3-none-any.whl.metadata (8.5 kB)
2024/11/10 09:21:47	Collecting soupsieve==2.6 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 68))
2024/11/10 09:21:47	  Downloading soupsieve-2.6-py3-none-any.whl.metadata (4.6 kB)
2024/11/10 09:21:57	Collecting stevedore==5.3.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 69))
2024/11/10 09:21:57	  Downloading stevedore-5.3.0-py3-none-any.whl.metadata (2.3 kB)
2024/11/10 09:22:02	Collecting subliminal==2.2.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 70))
2024/11/10 09:22:02	  Downloading subliminal-2.2.1-py3-none-any.whl.metadata (14 kB)
2024/11/10 09:22:06	Collecting text-unidecode==1.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 71))
2024/11/10 09:22:06	  Downloading text_unidecode-1.3-py2.py3-none-any.whl.metadata (2.4 kB)
2024/11/10 09:22:14	Collecting timeago==1.0.16 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 72))
2024/11/10 09:22:15	  Downloading timeago-1.0.16-py3-none-any.whl.metadata (4.5 kB)
2024/11/10 09:22:25	Collecting tmdbsimple==2.9.1 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 73))
2024/11/10 09:22:25	  Downloading tmdbsimple-2.9.1-py3-none-any.whl.metadata (6.9 kB)
2024/11/10 09:22:37	Collecting tornado==6.4 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 74))
2024/11/10 09:22:37	  Downloading tornado-6.4-cp38-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.5 kB)
2024/11/10 09:22:49	Collecting tus.py==1.3.4 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 75))
2024/11/10 09:22:49	  Downloading tus.py-1.3.4.tar.gz (3.8 kB)
2024/11/10 09:22:55	  Preparing metadata (setup.py): started
2024/11/10 09:22:56	  Preparing metadata (setup.py): finished with status 'done'
2024/11/10 09:22:59	Collecting tvdbsimple==1.0.6 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 76))
2024/11/10 09:22:59	  Downloading tvdbsimple-1.0.6.tar.gz (11 kB)
2024/11/10 09:23:01	  Preparing metadata (setup.py): started
2024/11/10 09:23:02	  Preparing metadata (setup.py): finished with status 'done'
2024/11/10 09:23:04	Collecting urllib3==2.2.3 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 77))
2024/11/10 09:23:04	  Downloading urllib3-2.2.3-py3-none-any.whl.metadata (6.5 kB)
2024/11/10 09:23:12	Collecting validators==0.34.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 78))
2024/11/10 09:23:12	  Downloading validators-0.34.0-py3-none-any.whl.metadata (3.8 kB)
2024/11/10 09:23:22	Collecting win-inet-pton==1.1.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 79))
2024/11/10 09:23:22	  Downloading win_inet_pton-1.1.0-py2.py3-none-any.whl.metadata (2.6 kB)
2024/11/10 09:23:29	Collecting wrapt==1.16.0 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 80))
2024/11/10 09:23:29	  Downloading wrapt-1.16.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.6 kB)
2024/11/10 09:23:34	Collecting xmltodict==0.14.2 (from -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 81))
2024/11/10 09:23:34	  Downloading xmltodict-0.14.2-py2.py3-none-any.whl.metadata (8.0 kB)
2024/11/10 09:23:39	ERROR: Could not find a version that satisfies the requirement zipp==3.20.12 (from versions: 0.1.0, 0.2.0, 0.2.1, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.4.0, 0.5.0, 0.5.1, 0.5.2, 0.6.0, 1.0.0, 1.1.0, 1.1.1, 1.2.0, 2.0.0, 2.0.1, 2.1.0, 2.2.0, 2.2.1, 3.0.0, 3.1.0, 3.2.0, 3.3.0, 3.3.1, 3.3.2, 3.4.0, 3.4.1, 3.4.2, 3.5.0, 3.5.1, 3.6.0, 3.7.0, 3.8.0, 3.8.1, 3.9.0, 3.9.1, 3.10.0, 3.11.0, 3.12.0, 3.12.1, 3.13.0, 3.14.0, 3.15.0, 3.16.0, 3.16.1, 3.16.2, 3.17.0, 3.18.0, 3.18.1, 3.18.2, 3.19.0, 3.19.1, 3.19.2, 3.19.3, 3.20.0, 3.20.1, 3.20.2)
2024/11/10 09:23:39	ERROR: No matching distribution found for zipp==3.20.12
2024/11/10 09:23:40	ERROR: Python package installation failed
2024/11/10 09:23:40	Installed modules:
2024/11/10 09:23:42	certifi==2024.8.30
2024/11/10 09:23:42	distlib==0.3.8
2024/11/10 09:23:42	filelock==3.16.1
2024/11/10 09:23:42	platformdirs==4.3.6
2024/11/10 09:23:42	six==1.16.0
2024/11/10 09:23:42	virtualenv==20.26.6
2024/11/10 09:23:42	End service_postinst
2024/11/10 09:23:42	install sickchill 20240301-11 End postinst ret=[0]
2024/11/10 09:23:45	install sickchill 20240301-11 Begin start-stop-status start
2024/11/10 09:24:05	install sickchill 20240301-11 End start-stop-status start ret=[1]
2024/11/10 09:24:10	(system) trigger sickchill 20240301-11 Begin start-stop-status stop
2024/11/10 09:24:10	(system) trigger sickchill 20240301-11 End start-stop-status stop ret=[0]
```

BKSteve commented:
@mreid-tt yeah had a little typo in the requirements file for zipp. Now zipp==3.21.0 should work.

hgy59 commented:
> Expecting a newer SickChill before requesting a release into the wild.

@miigotu is there a new release of sickchill on the way?

mreid-tt commented:
> @BKSteve, I've completed the rebase for you to resolve the qoriq build issue. Please complete the testing checklist to verify that the PR is ready for merging.

I think I may have messed up by doing a rebase on a foreign branch as noted by @hgy59. I now see this commit has changes across a number of packages as it appears to roll in the changes to the repo since. @hgy59, would this cause a huge issue when squashed and merged?

hgy59 commented:
@BKSteve do we really need unit testing in a release build? 
pytest, mock and others (flake)? (the * .dev dependencies) should be omitted.

hgy59 commented:
some findings of my test installation:

At package installation the call of `install_python_wheels` does not what I would expect.

All (cross compiled) wheels are downloaded and not installed from *.whl files in the package.
So installation is expected to fail on archs not having whl files to download (armv7, armv5, qoriq, i686)

For python packages that I maintain (homeassistant, octoprint) I never use `install_python_wheels` but this code:
```
    echo "Install packages from wheelhouse"
    pip install --disable-pip-version-check --no-deps --no-input --no-index ${SYNOPKG_PKGDEST}/share/wheelhouse/*.whl
```

And for the pure python modules (not in the package):
_using the cache is a bit tricky, we could also use `--no-cache-dir` but not the default cache dir of the user the installer runs with (root)._
```
    echo "Install pure python packages from index"
    pip install --disable-pip-version-check --no-deps --no-input --cache-dir ${PIP_CACHE_DIR} --requirement ${SYNOPKG_PKGDEST}/share/wheelhouse/requirements-pure.txt
```


The same solution is used in ffsync (syncstorage-rs) by @mreid-tt.


In current sickchill package, we could even call 
```
    pip install sickchill==2024.3.1
```
and create the package without python wheels.

I think if the sickchill python package dependencies are complete, it would be enough.

But my findings are, that the following are not installed this way
- attrs
- Deprecated
- Flask
- mock
- pluggy
- PyGithub
- PyJWT
- pymediainfo
- pyparsing
- pytest
- pyNACL
- wrapt

--- 

I think it is time for a redesign (or removal) of `install_python_wheels` in `mk/spksrc.service.installer.functions`

we should consider two options/versions for wheel installation
1. install crossenv wheels from whl files of the package and pure wheels from the index
2. install all wheels from whl files (for packages that include the pure wheels in the package for _"air gap"_ installation)

/cc @th0ma7

BKSteve commented:
> do we really need unit testing in a release build?
pytest, mock and others (flake)? (the * .dev dependencies) should be omitted.

No, probably don't need these dev related items. 

I was and still am somewhat out of my depth in regard spk building. I just took over a while back as nobody was doing it and got it to install with lots of assistance from the team here, especially on recent builds with the DSM 7 changes and deprecation forced by other packages.

> At package installation the call of `install_python_wheels` does not what I would expect.At package installation the call of install_python_wheels does not what I would expect.

> I think it is time for a redesign (or removal) of install_python_wheels in mk/spksrc.service.installer.functions
> we should consider two options/versions for wheel installation
>     1. install crossenv wheels from whl files of the package and pure wheels from the index  
>     2.  install all wheels from whl files (for packages that include the pure wheels in the package for "air gap" installation)

Many things are from looking at other packages and bringing them across, grateful for any advice pointing me in the right direction particularly on the question posed.

th0ma7 commented:
@hgy59 this is trickier than you might think... At package generation time a resulting `requirements.txt` file which combine ALL wheels (cross-compiled & pure) is being generated and packaged.  This was by design at the time so the installation would refer to a "unique" requirement file as it used to before I started reworking the wheel building process.  That I suspect may be worth having another look at considering all changes that cumulated since.

Further, in trying to save storage space online all pure packages where removed from the default packaging unless setting `WHEELS_PURE_PYTHON_PACKAGING_ENABLE` (that variable name is really too long...)

Using this current method, and considering pure python wheels are downloaded from the index, `pip` will always look at the index first then fallback to local.

This means for cross-compiled wheels, as joint under the unique `requirements.txt` file,  if you tested on x86_64 it will "always" download from the index if a pre-compiled version is available, then fallback using local package if available. 
 For almost every other cases on other arches it won't fnd anything using the index thus fallback to the local files.

Also note that (and to be confirmed), a downloaded `pyNACL` from idex for instance would contain the static library built-in, thus would not be using the shared lib that is shipped with the package.  That could be enough to enforce using local by default first, only then download from the index if unavailable.  Although I'm not sure there is a `pip` option for that, unless we split cross from pure as well within the resulting package.

hgy59 commented:
> > I think I may have messed up by doing a rebase on a foreign branch as noted by @hgy59. I now see this commit has changes across a number of packages as it appears to roll in the changes to the repo since. @hgy59, would this cause a huge issue when squashed and merged?
> 
> @mreid-tt it tells Merge remote-tracking branch 'origin/sickchill-20240202' into sickchill-20240202` but it sould be something like "Merge branch 'master' of https://github.com/SynoCommunity/spksrc into sickchill-20240202"

@BKSteve shall we rebase or start a new PR from scratch?

BKSteve commented:
> But my findings are, that the following are not installed this way
> attrs, Deprecated, Flask, mock, pluggy, PyGithub, PyJWT, pymediainfo, pyparsing, pytest, pyNACL, wrapt

During testing I put most of these in requirements.txt as it brought the dependencies sub dependants into the build as they didn't seem to be loading for all DSM HW. All my DSM are x64 so I relied on others to test on different hardware. 

For x64 devices it would probably just need sickchill== xxx and most dependencies and sub dependencies would be pulled in but some are in `speedups.dependencies` and one in particular `pymediainfo` was added recently specifically to req.txt as a really nice to have and even my x64 don't pull it but also don't want to make it an absolute necessity for all SickChill installs on other platforms. 
Now also considering adding `html5lib, Js2Py` and `flask-sqlalchemy` (frontend.dependencies) to requirements.txt as well.


But it's the other DSM HW that are the issue to build for and test to get a working package.

hgy59 commented:
@BKSteve can you please rebase with upstream/master? (or shall @mreid-tt or @hgy59 do this for you?)

we should get rid of the commits not related to this PR.

BKSteve committed:
rebase

BKSteve commented:
damn

hgy59 commented:
@BKSteve this has still failures at installation:

```
2024/11/14 00:28:59	ERROR: Cannot install -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 42), -r /volume1/@appstore/sickchill/share/wheelhouse/requirements.txt (line 54) and cryptography==41.0.3 because these package versions have conflicting dependencies.
2024/11/14 00:28:59	The conflict is caused by:
2024/11/14 00:28:59	    The user requested cryptography==41.0.3
2024/11/14 00:28:59	    jwt 1.3.1 depends on cryptography!=3.4.0 and >=3.1
2024/11/14 00:28:59	    pyopenssl 24.2.1 depends on cryptography<44 and >=41.0.5
2024/11/14 00:28:59	To fix this you could try to:
2024/11/14 00:29:00	1. loosen the range of package versions you've specified
2024/11/14 00:29:00	2. remove package versions to allow pip attempt to solve the dependency conflict
2024/11/14 00:29:00	ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
2024/11/14 00:29:31	ERROR: Python package installation failed
```

with the current requirements the build of ARMv7L still works.

--- 

I have a working version with 
- more recent modules in requirements
- additional pure requirements that seem mandatory (srt, pysubs2)
- one additional pure requirement (typing_extensions)
- charset_normalizer moved to crossenv requirements
- without dev dependencies (attrs, Deprecated, Flask, mock, pluggy, PyGithub, PyJWT)
- with different versions of greenlet (greenlet==3.1.0 for ARMv7L) and greenlet==3.1.1 for all others

Since this package does not support updates via github, I assume that PyGithub/PyJWT are obsolete

I could add a commit or provide the files for further discussion.

My local version runs as x64-7.1, aarch64-6.2.4 and armv7-7.1.

BKSteve commented:
Any inputs most appreciated

hgy59 committed:
sickchill: adjust requirements

- remove obsolete requirements
  - remove requirements not referenced by sickchill
  - including requirements for dev package and git installs
- update greenlet and add version specific requirements files
  - use greenlet==3.1.1 for DSM 7 (gcc >= 5)
  - limit greenlet to 3.1.0 for gcc < 5
- downgrade pyOpenSSL to 23.2.0 (compatible with cryptography==41.0.3)
- enable build for ARMv7L
- move charset_normalizer to crossenv requirements

BKSteve commented:
Note Mediainfo has been removed, although not a specific SickChill requirement if installed it helps read file data and is loaded through a `Try` so I recently put this in to improve rescan of files and particularly mp4 files.
```
try:
    from pymediainfo import MediaInfo as mediainfo
except (ModuleNotFoundError, RuntimeError):
    mediainfo = None
```
What would be best? Pure `pymediainfo==6.1.0` or in makefile `cross/mediainfo` based on the `media_info.py` code above?

hgy59 commented:
@BKSteve Ok, I understand

There are some extra features (optional) in Sickchill
- experimental
- speedups
- types
- frontend

When talking of pymediainfo I suppose we should include the _speedups_ extra feature.
and I guess we can ignore the experimantal feature.

Can you tell me what the _types_ and _frontend_ features are for?
I suppose the _frontend_ is like a different application, not required for sickchill (web) server.

hgy59 committed:
add requirements for optional "speedups" feature

BKSteve commented:
The front end is going to change in the future and is prep phase for future.
The types and experimental can be left out.

Thanks for the assistance. I've shared the link to builds for others to test via discord.

mreid-tt commented:
Superseded by #6517
